cmd/rule-evaluator/main.go

```diff
@@ -96,155 +96,110 @@ func main() {
 	// we reuse that constant.
 	version, err := export.Version()
 	if err != nil {
-		//nolint:errcheck
-		level.Error(logger).Log("msg", "Unable to fetch module version", "err", err)
+		_ = level.Error(logger).Log("msg", "Unable to fetch module version", "err", err)
 		os.Exit(1)
 	}
-	newExporter := exportsetup.FromFlags(a, fmt.Sprintf("rule-evaluator/%s", version))
 
-	notifierOptions := notifier.Options{Registerer: reg}
-
-	projectID := a.Flag("query.project-id", "Project ID of the Google Cloud Monitoring scoping project to evaluate rules against.").
-		Default(defaultProjectID).String()
-
-	targetURL := a.Flag("query.target-url", fmt.Sprintf("The address of the Prometheus server query endpoint. (%s is replaced with the --query.project-id flag.)", projectIDVar)).
-		Default(fmt.Sprintf("https://monitoring.googleapis.com/v1/projects/%s/location/global/prometheus", projectIDVar)).
-		String()
-
-	generatorURLStr := a.Flag("query.generator-url", "The base URL used for the generator URL in the alert notification payload. Should point to an instance of a query frontend that accesses the same data as --query.target-url.").
-		PlaceHolder("<URL>").String()
-
-	queryCredentialsFile := a.Flag("query.credentials-file", "Credentials file for OAuth2 authentication with --query.target-url.").
-		Default("").String()
-
-	disableAuth := a.Flag("query.debug.disable-auth", "Disable authentication (for debugging purposes).").
-		Default("false").Bool()
+	exporterOpts := export.ExporterOpts{
+		UserAgentProduct: fmt.Sprintf("rule-evaluator/%s", version),
+	}
+	exporterOpts.Default()
+	exportsetup.ExporterOptsFlags(a, &exporterOpts)
 
-	listenAddress := a.Flag("web.listen-address", "The address to listen on for HTTP requests.").
-		Default(":9091").String()
+	metadataOpts := exportsetup.MetadataOpts{}
+	metadataOpts.Default()
+	metadataOpts.SetupFlags(a)
 
-	configFile := a.Flag("config.file", "Prometheus configuration file path.").
-		Default("prometheus.yml").String()
+	haOpts := exportsetup.HAOptions{}
+	haOpts.Default()
+	haOpts.SetupFlags(a)
 
-	a.Flag("alertmanager.notification-queue-capacity", "The capacity of the queue for pending Alertmanager notifications.").
-		Default("10000").IntVar(&notifierOptions.QueueCapacity)
+	evaluatorOpts := evaluatorOptions{
```

Comment:
suggestion: Again, it seems a lot of the code from #1001 was folded in here, which makes this change hard to evaluate on its own merit.

Can we back this out, or perhaps get the other one merged first before proceeding with this review?

Reply:
Addressing in https://github.com/GoogleCloudPlatform/prometheus-engine/pull/1001